src/sentry/search/eap/spans.py

```diff
@@ -0,0 +1,253 @@
+                return TraceItemFilter(and_filter=AndFilter(filters=[resolved_lhs, resolved_rhs]))
+            else:
+                return TraceItemFilter(or_filter=OrFilter(filters=[resolved_lhs, resolved_rhs]))
+        elif resolved_lhs is None and resolved_rhs is not None:
+            return resolved_rhs
+        elif resolved_lhs is not None and resolved_rhs is None:
+            return resolved_lhs
+        else:
+            return None
+
+    def _resolve_terms(self, terms: event_filter.ParsedTerms) -> TraceItemFilter:
+        parsed_terms = []
+        for item in terms:
+            if isinstance(item, event_search.SearchFilter):
+                resolved_column, context = self.resolve_column(item.key.name)
+                if item.operator in constants.OPERATOR_MAP:
+                    operator = constants.OPERATOR_MAP[item.operator]
+                else:
+                    raise InvalidSearchQuery(f"Unknown operator: {item.operator}")
+                if isinstance(resolved_column.proto_definition, AttributeKey):
+                    parsed_terms.append(
+                        TraceItemFilter(
+                            comparison_filter=ComparisonFilter(
+                                key=resolved_column.proto_definition,
+                                op=operator,
+                                value=self._resolve_search_value(
+                                    resolved_column, item.operator, item.value.raw_value
+                                ),
+                            )
+                        )
+                    )
+                else:
+                    raise NotImplementedError("Can't filter on aggregates yet")
+            else:
+                raise NotImplementedError()
+        if len(parsed_terms) > 1:
+            return TraceItemFilter(and_filter=AndFilter(filters=parsed_terms))
+        else:
+            return parsed_terms[0]
+
+    def _resolve_search_value(
+        self,
+        column: ResolvedColumn,
+        operator: str,
+        value: str | int | datetime | Sequence[int] | Sequence[str],
+    ) -> AttributeValue:
+        column.validate(value)
+        if isinstance(column.proto_definition, AttributeKey):
+            column_type = column.proto_definition.type
+            if column_type == constants.STRING:
+                if operator in constants.IN_OPERATORS:
+                    if isinstance(value, list) and all(isinstance(item, str) for item in value):
+                        return AttributeValue(val_str_array=StrArray(values=value))
+                    else:
+                        raise InvalidSearchQuery(
+                            f"{value} is not a valid value for doing an IN filter"
+                        )
+                else:
+                    return AttributeValue(val_str=str(value))
+            elif column_type == constants.INT:
+                # These int casts are only necessary because floats aren't supported in the proto yet
```

Comment:
When the column type is `INT` then we should pass ints to the proto right? This comment is meant for float columns

Reply:
:) the search grammar currently turns any numeric field into a float, but because the proto currently only allows ints, i'm just passing them as ints to get things going